functions.py

```diff
@@ -1,35 +1,50 @@
 import csv
 import datetime
 import exceptions_tests
-import os #this is needed for using directory paths and manipulating them
-import sys
+import os
 from pip._vendor.distlib.compat import raw_input
 
+
 class Employee(object):
 
     def __init__(self, employee_id, name, phone, age):
+        """
+        This function gets the employee's id, name, phone and age from user.
+        employee_id = a string.
+        name = a string.
+        phone = a string.
+        age = a string.
+        """
         self.employee_id = employee_id
         self.name = name
         self.phone = phone
         self.age = age
 
-class Employees_list(object):
+
+class EmployeesList(object):
+
     def __init__(self, list_of_employees):
+        """
+        This function gets employees file's path from user.
+        list_of_employees = a string.
+        """
         self.list_of_employees = list_of_employees
 
     def add_employee(self, new_employee):
-        """
-        This function gets the credentials of a new employee from user and adds it to the employees file.
-        """
-        with open(self.list_of_employees, mode='a') as csvFile:
-            fileout = csv.writer(csvFile, delimiter=',', quoting=csv.QUOTE_ALL)
-            fileout.writerow([new_employee.employee_id, new_employee.name, new_employee.phone, new_employee.age])
+        new_employee_data = [new_employee.employee_id, new_employee.name, new_employee.phone, new_employee.age]
+        with open(self.list_of_employees, 'r+') as file1:
+            existing_lines = csv.reader(file1)
+            for row in existing_lines:
+                if new_employee_data in row:
```

Comment:
maybe check only for ID and not all the details -> 
if  new_employee_data[0] in row: 
---------------------------------------------
if it still doesn't work try to print the row and see what inside it....

Reply:
done